### PR TITLE
More engine related renaming and re-organizing

### PR DESCRIPTION
**Pull request by Berthalamew**
Try and organize some of the Engine related code we have into the Blam/Engine folder
rename structs so that they try and parody original struct names
move some of the structs/definitions to proper modules

This is beneficial since it tries to keep everything game related contained to a single folder and tries to parody the original layout and naming.

### CODE REVIEW

File: xlive/Blam/Engine/Simulation/MachineID.h [hunk 1]
```diff
@@ -2,5 +2,13 @@
 #pragma once
 struct s_machine_identifier
 {
-	char gap[6];
```
Comment (nukeulater) [inline xlive/Blam/Engine/Simulation/MachineID.h:5]: this is probably the abEnet, just do `byte machine_identifier[6]`

**Comment by nukeulater** [comment]
other than those unnecessary present in this PR that were reverted: https://github.com/pnill/cartographer/pull/426/commits/84b4b98b977cdb8d82a27247e2e1f46d009245f5 https://github.com/pnill/cartographer/pull/426/commits/3a454c901a03e0221808a19830a7888012799011

lgtm for the most part, and if @Himanshu-01 doesn't have anything to add.

File: xlive/Blam/Engine/game/local_players.h [hunk 2]
```diff
@@ -1,7 +1,38 @@
...
 
+struct s_player_identifier
+{
+	DWORD unk1;
+	DWORD unk2;
+};
+CHECK_STRUCT_SIZE(s_player_identifier, 8);
+
+#pragma pack(push, 1)
+struct s_local_player_properties
```
Comment (Himanshu-01) [inline xlive/Blam/Engine/game/local_players.h]: Actual name for this is `s_session_interface_globals` which comes under `networking` not `game `

File: xlive/H2MOD/Modules/Tweaks/Tweaks.cpp [hunk 3]
```diff
@@ -2,7 +2,7 @@
 
 #include "Tweaks.h"
 #include "Blam/Engine/game/game_time.h"
-#include "Blam\FileSystem\FiloInterface.h"
```
Comment (Himanshu-01) [inline xlive/H2MOD/Modules/Tweaks/Tweaks.cpp:5]: Would recommend maintaining a consistency during file inclusion 
Either use forward slashes `\` or or backward slashes  `/` for example this file has all the combinations messed up

```C
#include "Blam/Engine/tag_files/files_windows.h"
#include "Blam\Engine\Networking\NetworkMessageTypeCollection.h"
```

File: xlive/Blam/Engine/Simulation/MachineID.h (hunk 1 above)
Comment (Himanshu-01) [reply xlive/Blam/Engine/Simulation/MachineID.h:5]: Yea its actually abNet only but bungie used `s_machine_identfier` as their goto structure in a lot of places instead declaring a 6 byte long array everywhere so we should keep this structure in my opinion

File: xlive/H2MOD/Modules/Tweaks/Tweaks.cpp (hunk 3 above)
Comment (Himanshu-01) [reply xlive/H2MOD/Modules/Tweaks/Tweaks.cpp:5]: While you are at this . I would suggest getting rid of this namespace as well inside `objects/objects.h`

```c++
namespace Engine::Objects
{
	void create_new_placement_data(s_object_placement_data* object_placement_data, datum object_definition_idx, datum object_owner_idx, int a4);
	datum object_new(s_object_placement_data* object_placement_data);
	void apply_biped_object_definition_patches();
	void simulation_action_object_create(datum object_idx);
	void object_destroy(datum object_idx);
	int object_get_count();
	int object_count_from_iter();
}
```

Comment (Berthalamew) [reply xlive/H2MOD/Modules/Tweaks/Tweaks.cpp:5]: Will remove that namespace in another PR,

Comment (Berthalamew) [reply xlive/H2MOD/Modules/Tweaks/Tweaks.cpp:5]: > 
Changed includes to use forward slashes in tweaks

File: xlive/Blam/Engine/game/local_players.h (hunk 2 above)
Comment (Berthalamew) [reply xlive/Blam/Engine/game/local_players.h]: renamed, and moved updated definition to NetworkSession.h

File: xlive/Blam/Engine/Simulation/MachineID.h (hunk 1 above)
Comment (nukeulater) [reply xlive/Blam/Engine/Simulation/MachineID.h:5]: meant it should actually be defined as `byte machine_identifier[6]` in that structure

Comment (Berthalamew) [reply xlive/Blam/Engine/Simulation/MachineID.h:5]: done

**Pull request merged**